### DISCUSSION TITLE
Target velocity for IBM?

### DISCUSSION THREAD

**Discussion by AyushBoral** [xcompact3d/Incompact3d#226] [General]
How can we modify the target velocity at the immersed body boundary for IBM?

**Comment by slaizet** [comment]
Hi,

ubcx      ! Prescribed Velocity in X
ubcy      ! Prescribed Velocity in Y
ubcz      ! Prescribed Velocity in Z

See https://github.com/xcompact3d/Incompact3d/blob/master/examples/Cylinder/input_DNS300_LR_MOVING.i3d for an example.
Best,
Sylvain

**Reply by AyushBoral** [reply]
Thanks for the reply!! 
Actually, I am trying to impose rotation on the 3D cylinder case. Can you suggest the places where I have to modify in order to specify the tangential velocity instead of the default no-slip condition.
Thanks
Ayush Boral

**Comment by slaizet** [comment]
Hi,
I would suggest to do that with `iibm=1` and adapt the velocities in the `body `subroutine.
Best,
Sylvain

**Reply by AyushBoral** [reply]
Thank you very much
